csharp/BUILD

```diff
@@ -0,0 +1,33 @@
+# Protobuf C# runtime
+#
+# See also code generation logic under /src/google/protobuf/compiler/csharp.
+
+load("@rules_pkg//:mappings.bzl", "pkg_files", "strip_prefix")
+
+pkg_files(
+    name = "dist_files",
+    srcs = glob([
+        "keys/*",
+        "protos/*",
+        "src/**/*.cs*",  # .cs and .csproj
+    ]) + [
+        ".editorconfig",
```

Comment:
Is there a plan in place for how we will know whether to add new files to these packages?

Reply:
> Is there a plan in place for how we will know whether to add new files to these packages?

I don't have a complete answer right now. I think there are probably two considerations, which are related but separate:

1. What do we use to produce the release artifacts?
2. What do we want to validate about them in general?

I think the answer for (1) ultimately is that we want to use these rules. However, we need to find a strategy for vendoring googletest before that can happen.

For (2), we have only partial coverage today (C++ distcheck). It seems reasonable to me to change the non-Bazel-driven builds to unzip the distribution artifacts and run inside, though, which I think is the ground truth we would want.

So, for example, this:
https://github.com/protocolbuffers/protobuf/blob/main/tests.sh#L229-L234
could change to something like:
```
bazel build :protoc pkg:dist_java_tar
BAZEL_BIN=$(bazel info bazel-bin)
export PROTOC=$BAZEL_BIN/protoc
tar -C $dir -axf $BAZEL_BIN/pkg/dist_java_tar.tar.gz
cd $dir && $MVN ...
```

Reply:
Ok that makes sense. My main concern here is that someone adds a new file that is needed across distributions and it doesn't get included. For now, we should just make sure this is properly documented.

Reply:
Added comments to `Makefile.am`.